pkg/chunk/encoding/chunk.go

```diff
@@ -44,7 +44,7 @@ type Chunk interface {
 	// or a newly allocated version. In any case, take the returned chunk as
 	// the relevant one and discard the original chunk.
 	Add(sample model.SamplePair) ([]Chunk, error)
-	NewIterator() Iterator
+	NewIterator(Iterator) Iterator
```

Comment:
Can we have a comment here saying what the parameter means and what will happen to it?

Reply:
Added the same comment as in prometheus/tsdb